SWIG/volatilities.i

```diff
@@ -924,4 +925,281 @@ class KahaleSmileSectionPtr
+                          isBetaFixed, isNuFixed, isRhoFixed, isGammaFixed, vegaWeighted,
+                          endCriteria, method, dc));
+        }
+        Real alpha() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->alpha();
+        }
+        Real beta() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->beta();
+        }
+        Real nu() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->nu();
+        }
+        Real rho() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->rho();
+        }
+        Real rmsError() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->rmsError();
+        }
+        Real maxError() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->maxError();
+        }
+        EndCriteria::Type endCriteria() const {
+            return boost::dynamic_pointer_cast<ZabrInterpolatedSmileSection<Evaluation> >(*self)
+                ->endCriteria();
+        }
+    }
+};
+
+%enddef
+
+export_zabrinterpolatedsmilesection_curve(ZabrShortMaturityLognormalInterpolatedSmileSection, ZabrShortMaturityLognormal);
+export_zabrinterpolatedsmilesection_curve(ZabrShortMaturityNormalInterpolatedSmileSection, ZabrShortMaturityNormal);
+export_zabrinterpolatedsmilesection_curve(ZabrLocalVolatilityInterpolatedSmileSection, ZabrLocalVolatility);
+export_zabrinterpolatedsmilesection_curve(ZabrFullFdInterpolatedSmileSection, ZabrFullFd);
+
+%{
+typedef boost::shared_ptr<SmileSection> NoArbSabrSmileSectionPtr;
+typedef boost::shared_ptr<SmileSection> NoArbSabrInterpolatedSmileSectionPtr;
+%}
+
+%rename(NoArbSabrSmileSection) NoArbSabrSmileSectionPtr;
+class NoArbSabrSmileSectionPtr : public boost::shared_ptr<SmileSection> {
+  public:
+    %extend {
+        NoArbSabrSmileSectionPtr(
+               Time timeToExpiry, 
+               Rate forward,
+               const std::vector<Real> &sabrParameters,
+               const Real shift = 0.0) {
+            return new NoArbSabrSmileSectionPtr(
+                new NoArbSabrSmileSection(
+                          timeToExpiry,forward,sabrParameters,shift));
+        }
+        NoArbSabrSmileSectionPtr(
+               const Date &d, 
+               Rate forward,
+               const std::vector<Real> &sabrParameters,
+               const DayCounter &dc = Actual365Fixed(),
+               const Real shift = 0.0) {
+            return new NoArbSabrSmileSectionPtr(
+                new NoArbSabrSmileSection(
+                          d,forward,sabrParameters,dc,shift));
+        }
+        Real optionPrice(Rate strike,
```

Comment:
Are such downcasts for the virtual functions necessary here, or does the virtuality survive through SWIG?

Reply:
The method is inherited from `SmileSection`.  You don't need to redeclare it here.